### PR TITLE
add SPTAG algorithm

### PR DESCRIPTION
**Pull request by SueSu-Wish**
SPTAG: https://github.com/microsoft/SPTAG

### CODE REVIEW

File: ann_benchmarks/algorithms/sptag.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class Sptag(BaseANN):
+    def __init__(self, metric, algo='BKT'):
+        self._algo = str(algo)
+        self._metric = str(metric)
+
+    def fit(self, X):
+        self._sptag = SPTAG.AnnIndex(self._algo, 'Float', X.shape[1])
+	self._sptag.SetBuildParam("NumberOfThreads", '32')
```
Comment (erikbern) [inline ann_benchmarks/algorithms/sptag.py]: something is off with the indentation here

**Comment by erikbern** [comment]
Nice! It looks like building boost is very slow though. Is there any way this can be done with a standard version of boost?

https://travis-ci.org/erikbern/ann-benchmarks/builds/545044876?utm_source=github_status&utm_medium=notification

I'll merge this for now though!

**Comment by erikbern** [comment]
thanks @SueSu-Wish!

**Comment by SueSu-Wish** [comment]
You are welcome. : ) I guess it can't be done with a standard version of boost.

**Comment by erikbern** [comment]
can't you just install boost 1.67 using the debian package? https://packages.debian.org/sid/libboost1.67-all-dev

i can try later this week and see if it works

**Pull request merged**